planner/core/logical_plan_builder.go

```diff
@@ -1169,14 +1169,24 @@ func (g *gbyResolver) Leave(inNode ast.Node) (ast.Node, bool) {
 			return inNode, false
 		}
 	case *ast.PositionExpr:
-		if v.N < 1 || v.N > len(g.fields) {
-			g.err = errors.Errorf("Unknown column '%d' in 'group statement'", v.N)
+		var pos int64
+		var isPos bool
+		if val, ok := v.N.(*driver.ValueExpr); ok {
+			pos, isPos = val.GetValue().(int64)
+			if isPos && (pos < 1 || pos > int64(len(g.fields))) {
+				g.err = errors.Errorf("Unknown column '%d' in 'group statement'", pos)
+			}
+		}
+		if !isPos {
+			g.err = errors.Errorf("Unknown column '%+v' in 'group statement'", v.N.Text())
+		}
```

Comment:
MySQL and TiDB now are consistent for raising no error in this case:
```
mysql> prepare stmt from 'select ?, sum(a) from t group by ?';
Query OK, 0 rows affected (0.00 sec)
Statement prepared
```
would this change break the consistency?

Reply:
Thank you for the error case. I'll come back after fixing the issue.